### DISCUSSION TITLE
How to setup mono repo with Chakra?

### DISCUSSION THREAD

**Discussion by wileymc** [chakra-ui/chakra-ui#6692] [General]
We have a monorepo using turbo repo with the basic structure:
/apps
../app1
 ../app2
/packages
../ui
 ..../ComponentName
 ......ComponentName.tsx

Both app1 and app2 are wrapped in a <ChakraProvider> and we want to define some reusable (across app1 and app2) components in packages/ui. @chakra is being imported in different package.json files in app1, app2, and packages/ui.  However, chakra components defined in packages/ui and pulled into app1 and app2 are not being styled at all.  Does anyone have a similar setup that is working properly?

**Comment by carlbarrdahl** [comment]
Here's what I usually do:

1. Set up project with Turborepo (`npx create-turbo@latest`)
2. Install ChakraUI to `/packages/ui`

```tsx
// packages/ui/index.tsx
import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { PropsWithChildren } from "react";

// Export all components from Chakra UI so they can be imported from app eg:
// import { Button } from "ui"
export * from "@chakra-ui/react";

// You can also export icons here
// export * from "@chakra-ui/icons";

const theme = extendTheme({ });

export default function ThemeProvider(props: PropsWithChildren) {
  return <ChakraProvider theme={theme} {...props} />;
}
```

```tsx
// apps/web/pages/_app.tsx
import { AppProps } from "next/app";
import ThemeProvider from "ui";

function App({ Component, pageProps }: AppProps) {
  return (
    <ThemeProvider>
      <Component {...pageProps} />
    </ThemeProvider>
  );
}

export default App
```

**Reply by wileymc** [reply]
Awesome thank you for this writeup!
